### PR TITLE
keep session after login with email

### PR DESCRIPTION
**Pull request by AvshalomP**

### CODE REVIEW

File: src/pages/Auth/SignIn/EmailLoginSection.tsx [hunk 1]
```diff
@@ -9,16 +9,27 @@ import { COLORS } from '@constants/Colors';
...
 
   const onHandleSubmit = async (submitObj: EmailSignInObj): Promise<void> => {
+    setLoginErrMsg('');
+    setIsLoading(true);
+
     const authRes = await authManager.emailSignIn(submitObj.email, submitObj.password);
-    authRes.authState === AuthState.EMAIL_SIGNIN_SUCCESS && dispatch(setSignInAuthInfo(authRes.res));
+
+    if (authRes.authState === AuthState.EMAIL_SIGNIN_SUCCESS) {
+      dispatch(setAuthInfo(authRes.res));
```
Comment (asandler77) [inline src/pages/Auth/SignIn/EmailLoginSection.tsx:28]: should be isLoading set to false here?

File: src/pages/Auth/SignUp/EmailSignUpSection.tsx [hunk 2]
```diff
@@ -9,18 +9,28 @@ import { SIGN_UP } from '@pages/Auth/SignUp/consts';
...
   const onHandleSubmit = async (submitObj: EmailSignUpObj): Promise<void> => {
+    setSignupErrMsg('');
+    setIsLoading(true);
+
     const userInfo = { displayName: `${submitObj.firstName} ${submitObj.lastName}` };
     const authRes = await authManager.emailSignUp(submitObj.email, submitObj.password, userInfo);
 
-    authRes.authState === AuthState.EMAIL_SIGNUP_SUCCESS && dispatch(setSignUpAuthInfo(authRes.res));
+    if (authRes.authState === AuthState.EMAIL_SIGNUP_SUCCESS) {
+      dispatch(setAuthInfo(authRes.res));
```
Comment (asandler77) [inline src/pages/Auth/SignUp/EmailSignUpSection.tsx:29]: should be isLoading set to false here?

File: src/pages/Auth/ForgotPass/EmailPasswordUpdateSection.tsx [hunk 3]
```diff
@@ -16,10 +16,21 @@ import { AuthScreenNavigationProp } from '@navigation/AuthNavigator';
...
+  const [forgotErrMsg, setForgotErrMsg] = useState('');
 
   const onHandleSubmit = async (submitObj: ResetPasswordObj): Promise<void> => {
+    setForgotErrMsg('');
+    setIsLoading(true);
+
     const resetPassRes = await authManager.resetPassword(submitObj.email);
-    resetPassRes === AuthState.RESET_PASSWORD_SUCCESS && navigate(AuthStackScreensNames.SIGN_IN);
+
+    if (resetPassRes === AuthState.RESET_PASSWORD_SUCCESS) {
```
Comment (asandler77) [inline src/pages/Auth/ForgotPass/EmailPasswordUpdateSection.tsx:28]: should be isLoading set to false here?

Comment (AvshalomP) [reply src/pages/Auth/ForgotPass/EmailPasswordUpdateSection.tsx:28]: in this case yes. thanks

File: src/pages/Auth/SignUp/EmailSignUpSection.tsx (hunk 2 above)
Comment (AvshalomP) [reply src/pages/Auth/SignUp/EmailSignUpSection.tsx:29]: no as we are switching to a different stack, so this component is being unmounted

File: src/pages/Auth/SignIn/EmailLoginSection.tsx (hunk 1 above)
Comment (AvshalomP) [reply src/pages/Auth/SignIn/EmailLoginSection.tsx:28]: no as we are switching to a different stack, so this component is being unmounted

File: src/navigation/AppNavigator.tsx [hunk 4]
```diff
@@ -18,6 +20,23 @@ export type RootStackParamList = {
...
+  const onAuthStateChanged = (user: FirebaseAuthTypes.User | null): void => {
+    user ? dispatch(setAuthInfo(user)) : dispatch(clearAuthInfo());
+    if (initializing) {
+      setInitializing(false);
+    }
+  };
+
+  useEffect(() => {
+    const subscriber = auth().onAuthStateChanged(onAuthStateChanged);
+    return subscriber; // unsubscribe on unmount
```
[the remark this answers is not in the document]
Comment (AvshalomP) [reply src/navigation/AppNavigator.tsx:37]: it's the same and whoever wants to add something else on unmount, will create a function

**Review by asandler77 — APPROVED** [review]

**Pull request merged**